docs/design/proposals/non-graceful-node-shutdown.md

```diff
@@ -0,0 +1,127 @@
+# Non graceful node shutdown
+
+In Kubernetes, when a node becomes unhealthy or is intentionally drained,
+taints `node.kubernetes.io/out-of-service` may be applied to mark the node
+as unavailable. This results in the forceful deletion of pods scheduled on
+the node and the cleanup of associated VolumeAttachment objects.
+However, this forceful cleanup sequence bypasses key CSI lifecycle calls most
+notably, the `NodeUnpublishVolume` and `NodeUnstageVolume` entry points are not
+invoked prior to `ControllerUnpublishVolume` call.
+
+## Problem
+
+When `ControllerUnpublishVolume` is called without the node first having cleaned
+up the volume (via `NodeUnstageVolume` and `NodeUnpublishVolume`), the CSI driver
+has no opportunity to revoke the node’s access to the volume.The node may still
+hold active mounts, open file handles, or client sessions. This can lead to data
+corruption due to writes from disconnected yet still-active clients.
```

Comment:
_disconnected_ isn't really correct, as a disconnected client can not reach the storage. A client may re-connect later on, causing havoc. Or, the node may be somehow broken, but apps may still be running, this is the major concern.